src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,260 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Net.Http
+{
+    public sealed class HttpEnvironmentProxyCredientials : ICredentials
+    {
+        // Wrapper class for cases when http and https has different authentication.
+        private readonly NetworkCredential _http;
+        private readonly NetworkCredential _https;
+
+        public HttpEnvironmentProxyCredientials(NetworkCredential http, NetworkCredential https)
+        {
+            _http = http;
+            _https = https;
+        }
+
+        public NetworkCredential GetCredential(Uri uri, string authType)
+        {
+            return uri.Scheme == Uri.UriSchemeHttp ? _http : _https;
```

Comment:
The only Uris that should be valid here are literally the two Uris you return from GetProxy.  So it's probably better to compare the passed uri to exactly those uris.  For one, it will almost always be the same Uri instance, so the == check is cheap.  For another, if for some bizarre reason a user passes some other Uri here, it's better to return null.